src/Symfony/Component/Yaml/Parser.php

```diff
@@ -221,7 +221,14 @@ private function doParse($value, $flags)
                 try {
                     Inline::$parsedLineNumber = $this->getRealCurrentLineNb();
                     $i = 0;
-                    $key = Inline::parseScalar($values['key'], 0, null, $i, !(Yaml::PARSE_KEYS_AS_STRINGS & $flags));
+                    $evaluateKey = !(Yaml::PARSE_KEYS_AS_STRINGS & $flags);
```

Comment:
I really don't get why we would ever want to use this flag, this is like saying "I don't want to respect the spec".